### DISCUSSION TITLE
Custom array attribute name resolution issue

### DISCUSSION THREAD

**Discussion by developman-akl** [livewire/livewire#3824] [7. Bugs (archived)] [answered]
Laravel Version: 8.61.0
Livewire version: 2.6.5.
PHP Version: 7.4

### Description:
Custom `:attribute` names are not fetched anymore from the `attributes` array of the `resources/lang/xx/validation.php` language file. It was working fine in version **2.5.5.**

Maybe related to the changes in 2.6.0.:
[Add support for rules ending in array with nested rule under array](https://github.com/livewire/livewire/commit/f4939ee0561bacbd0e8b255e1b7e84d8306611bd)

### Steps To Reproduce:
The rule:
```
public function rules()
{
    return [
        'lines.*.darabszam' => [
            'integer',
            'required',
            'min:1',
            'max:999999999999',
        ]
    ],
}
```

`resources/lang/hu/validation.php`
```
'attributes' => [
    'lines.*.darabszam' => 'Darabszám',
]
```

```
'custom' => [
    'lines.*.darabszam' => [
        'min' => 'A(z) :attribute -nak legalább ennyinek kell lennie: :min.',
    ]
]
```

The validator data called from `updated()`:
![image](https://user-images.githubusercontent.com/48017129/134094700-d340c840-66aa-417f-8533-426fc1a5b69b.png)

![image](https://user-images.githubusercontent.com/48017129/133983253-6ccee58c-1f35-4b30-b9cb-0d6187ab8d2c.png)

It was still working fine in version 2.5.5.
![image](https://user-images.githubusercontent.com/48017129/133983554-cda2bf09-d399-4acb-a8f6-6c63d8a3ddb5.png)

**Comment by developman-akl** [comment]
It is definietly an issue since 2.6.0.

**Comment by Mecit** [comment]
I'm also facing the same issue. The problem seems to be caused by the `validateOnly` method. When I submit a form wholly and call the `validate` method, the custom attributes are parsed correctly. 

Imagine I have a custom attribute set as follows:

```
    'attributes' => [
        'terms.*.starts_at' => 'start date',
    ],
```

This works as expected:
```
public function save()
{
    $this->validate(); // Yields "The start date field is required."
   ...
```

This doesn't:
```
public function updated($field)
{
    $this->validateOnly($field); // Yields "The terms.1.start at field is required."
   ...
```

Any updates on this would be appreciated.

**Comment by developman-akl** [comment]
@calebporzio Coud you please give some ETA on this fix?
This bug literally wiped out a stock Laravel function.

**Comment by joshhanley** [comment]
@developman-akl and @Mecit sorry I'm not sure how I missed this!

I currently don't have time to look at this until the end of the week at the earliest, but if either of you can work out what is causing it, I'd be happy to fix it sooner, or PR's to fix it are also welcome!

Thanks!

**Reply by developman-akl** [reply]
Thank you looking into this. What I have learned about the root cause it is related to the changes introduced in 2.6.0.:
[Add support for rules ending in array with nested rule under array
](https://github.com/livewire/livewire/commit/f4939ee0561bacbd0e8b255e1b7e84d8306611bd)
It's working fine in version 2.5.5.

**Comment by joshhanley** [comment]
@developman-akl and @Mecit I have investigated this issue and it turns out it was actually this PR #3596 that has caused the issue.

I'm currently trying to fix this custom attributes issue and also get the PR to work as intended.
But if not, we will revert that PR so that way custom attributes are working again.

Hope this helps!

**Reply by developman-akl** [reply]
@joshhanley Is there any update re this fix? Cheers!

**Reply by joshhanley** [reply]
@developman-akl I've got a solution running, but have run into some edge cases that I am trying to work through. But will either get this fix, or revert the problem code, in the next release 🙂

**Comment by joshhanley** [comment]
@developman-akl and @Mecit finally got some time to finish this PR today, so it has been submitted, see PR #4188.

Hope this helps!

**Comment by joshhanley** [comment, marked as answer]
@developman-akl fix has been merged, if you get a chance, can you pull in master and test to make sure it's all fixed for you?

Thanks!

**Reply by Mecit** [reply]
Sorry for the late feedback. With the 2.8.1 release, everything is back to normal, now. Thanks a lot, @joshhanley.

**Reply by joshhanley** [reply]
@Mecit no worries! Thanks for testing 🙂

**Reply by developman-akl** [reply]
@joshhanley Yes it works, thank you!

**Reply by joshhanley** [reply]
@developman-akl great! 😁 No worries!
